src/apps/js_generic/js_generic.cpp

```diff
@@ -185,6 +186,51 @@ namespace ccfapp
     return JS_NewArrayBufferCopy(ctx, key.data(), key.size());
   }
 
+  static JSValue js_generate_rsa_key(
+    JSContext* ctx, JSValueConst, int argc, JSValueConst* argv)
+  {
+    if (argc != 1 && argc != 2)
+      return JS_ThrowTypeError(
+        ctx, "Passed %d arguments, but expected 1 or 2", argc);
+
+    int32_t key_size, key_exponent;
+    if (JS_ToInt32(ctx, &key_size, argv[0]) < 0)
+    {
+      js_dump_error(ctx);
+      return JS_EXCEPTION;
+    }
+
+    if (argc == 2 && JS_ToInt32(ctx, &key_exponent, argv[1]) < 0)
+    {
+      js_dump_error(ctx);
+      return JS_EXCEPTION;
+    }
+
+    // Supported key sizes for RSA.
+    if (key_size < 1024 || key_size >= MBEDTLS_MPI_MAX_BITS)
```

Comment:
This file shouldn't know about mbedtls. Better to move this check into a function `is_supported_rsa_key_size(n)` in the crypto namespace.

Reply:
mbedTLS has tighter limits, but all of these checks are unnecessary anyways. I just added a few of them to catch common problems, but I'll remove them and just return whatever `make_rsa_key_pair` throws.